modules/nf-core/kma/kma/main.nf

```diff
@@ -0,0 +1,66 @@
+process KMA {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/kma:1.4.15--he4a0461_0' :
+        'biocontainers/kma:1.4.15--he4a0461_0' }"
+
+    input:
+    tuple val(meta) , path(reads)
+    tuple val(meta2), path(index)
+    val interleaved
+    val mat_format
+
+    output:
+    tuple val(meta), path("*.res"),     optional: true, emit: res
+    tuple val(meta), path("*.fsa"),     optional: true, emit: fsa
+    tuple val(meta), path("*.aln"),     optional: true, emit: aln
+    tuple val(meta), path("*.frag.gz"), optional: true, emit: frag
+    tuple val(meta), path("*.mat.gz"),  optional: true, emit: mat   // if mat_format == true
+    tuple val(meta), path("*.spa"),     optional: true, emit: spa   // if ext.args contains '-Sparse' (only output in this case)
```

Comment:
why are all of them marked as optional?

Reply:
This is because of the tool's `-Sparse` option. {aln, res, frag, fsa} are returned in default mode. If the `-Sparse` option is used then only {spa} is returned. {mat} is in any of the former cases another optional output.

Reply:
ok then we can leave it like that :) but the stub needs to reflect that then (there you can check what is handed to the module and only create the file that you would expect)

Reply:
it seems you follow the Harshil Alignment style. If yes, you should align the commas not the word `optional`: [Comma Example](https://nf-co.re/docs/contributing/code_editors_and_styling/harshil_alignment#:~:text=Comma-,Example,-%E2%9D%8C%20Bad)